### PR TITLE
Remove pre1254 flag for compressor; 9 years is enough time for deprecation

### PR DESCRIPTION
**Pull request by ArneBab**

### CODE REVIEW

File: src/freenet/support/compress/OldLZMACompressor.java [hunk 1]
```diff
@@ -30,6 +30,7 @@ public void shouldUpdate(){
 	// Copied from EncoderThread. See below re licensing.
 	@Override
 	public Bucket compress(Bucket data, BucketFactory bf, long maxReadLength, long maxWriteLength) throws IOException, CompressionOutputSizeException {
+		Logger.warning(this, "OldLZMA compression is buggy and no longer supported. It only exists to allow reinserting keys.");
```
Comment (nextgens) [inline src/freenet/support/compress/OldLZMACompressor.java]: I'd add a @deprecated decorator too

File: src/freenet/support/compress/OldLZMACompressor.java [hunk 2]
```diff
@@ -52,6 +53,7 @@ public Bucket compress(Bucket data, BucketFactory bf, long maxReadLength, long m
 
 	@Override
 	public long compress(InputStream is, OutputStream os, long maxReadLength, long maxWriteLength) throws IOException, CompressionOutputSizeException {
+		Logger.warning(this, "OldLZMA compression is buggy and no longer supported. It only exists to allow reinserting keys.");
```
Comment (nextgens) [inline src/freenet/support/compress/OldLZMACompressor.java]: same here

File: src/freenet/support/compress/OldLZMACompressor.java (hunk 1 above)
Comment (ArneBab) [reply src/freenet/support/compress/OldLZMACompressor.java]: good point — thank you!

**Comment by ArneBab** [comment]
I now added a test for decompressing data compressed with LZMA (old). See https://github.com/freenet/fred/pull/680/files#diff-1884d76e1d3041f97138e6fb2c3f638e

File: src/freenet/support/compress/NewLZMACompressor.java [hunk 3]
```diff
@@ -66,6 +66,9 @@ public Bucket compress(Bucket data, BucketFactory bf, long maxReadLength, long m
 	public long compress(InputStream is, OutputStream os, long maxReadLength, long maxWriteLength,
 						 final long amountOfDataToCheckCompressionRatio, final int minimumCompressionPercentage)
 			throws IOException, CompressionRatioException {
+		if (amountOfDataToCheckCompressionRatio == Long.MAX_VALUE && minimumCompressionPercentage == 0) {
+		  return compress(is, os, maxReadLength, maxWriteLength);
```
[the remark this answers is not in the document]
Comment (ArneBab) [reply src/freenet/support/compress/NewLZMACompressor.java]: This was a mistake corrected later. Sorry — will be unmade with the merge of https://github.com/freenet/fred/pull/679/files

File: src/freenet/support/compress/OldLZMACompressor.java (hunk 2 above)
Comment (ArneBab) [reply src/freenet/support/compress/OldLZMACompressor.java]: added

**Comment by ArneBab** [comment]
There are test failures → not yet ready for merge.

**Pull request merged**